### PR TITLE
ZEN-33016 New zenhubworker service for flexible per-customer tuning

### PR DESCRIPTION
**Pull request by romanbakaleyko**

### CODE REVIEW

File: migrations/src/zenservicemigration/migrations/add_zenhubworker_user.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    return changed
+
+
+def _deploy_zenhub_worker_user(ctx):
+    """
+    deploy zenhubworker (user) service
+    """
+
+    zh = next((s for s in ctx.services if s.name == "zenhub"), None)
```
Comment (jpeacock-zenoss) [inline migrations/src/zenservicemigration/migrations/add_zenhubworker_user.py:44]: While uncommon, this code should handle the situation of multiple zenhub services deployed (i.e. multiple hubs).

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

File: migrations/src/zenservicemigration/migrations/add_zenhubworker_user.py (hunk 1 above)
Comment (romanbakaleyko) [reply migrations/src/zenservicemigration/migrations/add_zenhubworker_user.py]: fixed

File: migrations/src/zenservicemigration/migrations/add_zenhubworker_user.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    # Get root service for its ImageID value
+    zproxy = ctx.getTopService()
+    template["ImageID"] = zproxy.imageID
+
+    # Add zenhubworker for each zenhub.
+    for hub in hubs:
+        folder = ctx.getServiceParent(hub)
+        ctx.deployService(json.dumps(template), folder)
+        print("zenhubworker (user) service succesfully added")
```
Comment (jpeacock-zenoss) [inline migrations/src/zenservicemigration/migrations/add_zenhubworker_user.py:62]: There are three 's' in 'successfully'.

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

File: services/Zenoss.cse/Zenoss/Collection/localhost/zenhub-server.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      - "EventService:sendEvent"
+      - "EventService:sendEvents"
+
+adm:
+    spec: "Products.ZenHub.server.executors:WorkerPoolExecutor"
+    routes:
+      - "*:applyDataMaps"
+
+user:
+    spec: "Products.ZenHub.server.executors:WorkerPoolExecutor"
```
Comment (jpeacock-zenoss) [inline services/Zenoss.cse/Zenoss/Collection/localhost/zenhub-server.yaml:18]: This file is not needed.

File: services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json [hunk 4]
```diff
@@ -16,6 +16,11 @@
             "FileName": "/opt/zenoss/etc/zenhub.conf",
             "Owner": "zenoss:zenoss",
             "Permissions": "0664"
+        },
+        "/opt/zenoss/etc/zenhubservicecall.conf": {
+            "FileName": "/opt/zenoss/etc/zenhubservicecall.json",
```
Comment (jpeacock-zenoss) [inline services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json:21]: These should be `/opt/zenoss/etc/zenhub-server.yaml`.

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

File: services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json (hunk 4 above)
Comment (romanbakaleyko) [reply services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json]: done

File: services/Zenoss.cse/Zenoss/Collection/localhost/zenhub-server.yaml (hunk 3 above)
Comment (romanbakaleyko) [reply services/Zenoss.cse/Zenoss/Collection/localhost/zenhub-server.yaml]: done

File: services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json [hunk 5]
```diff
@@ -16,6 +16,11 @@
             "FileName": "/opt/zenoss/etc/zenhub.conf",
             "Owner": "zenoss:zenoss",
             "Permissions": "0664"
+        },
+        "/opt/zenoss/etc/zenhubservicecall.conf": {
```
Comment (jpeacock-zenoss) [inline services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json:20]: The convention has been to have the key and filename consistent with each other so this key should should also read `/opt/zenoss/etc/zenhub-server.yaml`.

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

**Comment by romanbakaleyko** [comment]
test this

File: services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json (hunk 5 above)
Comment (romanbakaleyko) [reply services/Zenoss.cse/Zenoss/Collection/localhost/zenhub/service.json]: done

**Review by jpeacock-zenoss — APPROVED** [review]

**Pull request merged**